corehq/apps/es/transient_util.py

```diff
@@ -0,0 +1,103 @@
+"""Transient utilities needed during the interim of converting HQ Elastic logic
+to use the new "client adapter" models.
+"""
+
+from .client import ElasticDocumentAdapter
+from .registry import get_registry
+
+
+def get_adapter_mapping(adapter):
+    """Temporary function for fetching the Elastic mapping (still defined in
+    pillowtop module) for an adapter.
+    """
+    return _DOC_MAPPINGS_BY_INDEX[(adapter.index_name, adapter.type)]
+
+
+def from_dict_with_possible_id(doc):
+    """Temporary "common" function for adapters who don't yet own their document
+    ``from_python()`` logic.
+    """
+    if "_id" in doc:
+        return doc["_id"], {k: doc[k] for k in doc if k != "_id"}
+    return None, doc
+
+
+def doc_adapter_from_info(index_info, for_export=False):
+    """Return the document adapter for the provided ``index_info`` object.
+
+    :param index_info: instance of a pillowtop ``ElasticsearchIndexInfo`` object
+    :param for_export: ``bool`` used to instantiate the adapter instance
+    :returns: instance of an ``ElasticDocumentAdapter`` subclass
+    """
+    return _DOC_ADAPTERS_BY_INDEX[index_info.index](for_export=for_export)
+
+
+def doc_adapter_from_alias(index_alias, for_export=False):
+    """Return the document adapter for the provided ``index_alias``.
+
+    :param index_alias: ``str`` name of a valid alias assigned to an index
+    :param for_export: ``bool`` used to instantiate the adapter instance
+    :returns: instance of an ``ElasticDocumentAdapter`` subclass
+    """
+    return _DOC_ADAPTERS_BY_ALIAS[index_alias](for_export=for_export)
+
+
+def report_and_fail_on_shard_failures(search_result):
+    ElasticDocumentAdapter._report_and_fail_on_shard_failures(search_result)
+
+
+def _populate_doc_adapter_map(is_test):
```

Comment:
nit: I just noticed that this function is only called outside this file. The `_` naming convention is used to indicate private functions that are not intended to be called outside the module. Since this is exclusively used outside the module, `populate_doc_adapter_map` would be a clearer name.

Reply:
Yes, I did that intentionally to discourage its use anywhere else. In hindsight this certainly has a smell. I'll remove the `_` and add a docstring instead.